src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReference.kt

```diff
@@ -12,6 +12,10 @@ import org.rust.lang.core.types.BoundElement
 interface RsPathReference : RsReference {
     override fun getElement(): RsPath
 
+    fun resolvePublicOnly(): RsElement? = multiResolvePublicOnly().singleOrNull()
```

Comment:
I think `publicOnly` is a confusing name since as I understand it resolves visible items that can have different visibility

Reply:
Good point, what do you think about `resolveCheckVisibility`?

Reply:
Replaced to `resolveIfVisible`